src/gax-internal/src/host.rs

```diff
@@ -20,20 +20,29 @@ use std::str::FromStr;
 ///
 /// Notably, locational and regional endpoints are detected and used as the
 /// host. For VIPs and private networks, we need to use the default host.
-pub(crate) fn host_from_endpoint(
+///
+/// Accepting a generic function as a parameter lets us avoid dead code warnings
+/// when only one of HTTP, gRPC is enabled.
+///
+/// We could have returned a boolean that says whether to use the default
+/// endpoint vs. the custom endpoint. Note, however, that:
+///   (1) this function already computes the origin and host.
+///   (2) consumers of this function would have to recompute the values.
```

Comment:
This is very weird, but I cannot think of a better approach in 10 minutes (something with a type that is convertible to `Uri` or `String`? Two different functions with the thing returning the tuple as a common implementation detail for both?)

I am fine with approving because the code in `pub(crate)`.

Reply:
> This is very weird

Agree. We ugly up the code for a micro-optimization.

> Two different functions with the thing returning the tuple as a common implementation detail for both?

I originally had it as a type, but the compiler detected that one of the fields was not read when only one of HTTP / gRPC is enabled.

Also the two different functions would have to be feature gated. And if we are going to do that, might as well put the different functions in `http.rs` / `grpc.rs`.